code/modules/library/lib_items.dm

```diff
@@ -26,13 +28,46 @@
 	max_integrity = 200
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, FIRE = 50, ACID = 0)
 	var/state = BOOKCASE_UNANCHORED
-	/// When enabled, books_to_load number of random books will be generated for this bookcase when first interacted with.
+	/// When enabled, books_to_load number of random books will be generated for this bookcase
 	var/load_random_books = FALSE
 	/// The category of books to pick from when populating random books.
 	var/random_category = null
 	/// How many random books to generate.
 	var/books_to_load = 0
 
+/obj/structure/bookcase/Initialize(mapload)
+	. = ..()
+	if(!mapload || QDELETED(src))
+		return
+	set_anchored(TRUE)
+	state = BOOKCASE_FINISHED
+	for(var/obj/item/I in loc)
+		if(!isbook(I))
+			continue
+		I.forceMove(src)
+	update_appearance()
+	SSlibrary.shelves_to_load += src
+
+///Loads the shelf, both by allowing it to generate random items, and by adding its contents to a list used by library machines
+/obj/structure/bookcase/proc/load_shelf()
+	//Loads a random selection of books in from the db, adds a copy of their info to a global list
+	//To send to library consoles as a starting inventory
+	if(load_random_books)
+		create_random_books(books_to_load, src, FALSE, random_category)
+		update_appearance() //Make sure you look proper
+
+	var/area/our_area = get_area(src)
+	var/area_type = our_area.type //Save me from the dark
+
+	if(!GLOB.roundstart_books_by_area[area_type])
+		GLOB.roundstart_books_by_area[area_type] = list()
+
+	//Time to populate that list
+	var/list/books_in_area = GLOB.roundstart_books_by_area[area_type]
+	for(var/obj/item/book/book in contents)
```

Comment:
why are you searching for a single typepath of books here when there are 3 parent typepaths for books

Reply:
Because I only care about books with info datums, which bibles and cutouts don't have. Also because create_random_books doesn't generate anything but standard books.